custom_components/unifiprotect/number.py

```diff
@@ -72,35 +89,34 @@ async def async_setup_entry(
     if not protect_data.data:
         return
 
-    number_entities = []
-    for item, item_type in NUMBER_TYPES.items():
-        required_field = item_type[_NUMBER_REQUIRES]
+    entities = []
+
+    for description in NUMBER_TYPES:
         for device_id in protect_data.data:
-            if protect_data.data[device_id].get("type") in item_type[_NUMBER_TYPE]:
-                if required_field and not protect_data.data[device_id].get(
-                    required_field
-                ):
+            if protect_data.data[device_id].get("type") in description.ufp_device_type:
```

Comment:
You can save some indent here by reversing the conditional and doing a `continue`, then outdent below